presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoResultSet.java

```diff
@@ -1793,6 +1798,14 @@ private ResultsPageIterator(StatementClient client, Consumer<QueryStats> progres
 
             return endOfData();
         }
+
+        private void checkInterruption()
+        {
+            if (Thread.currentThread().isInterrupted()) {
+                client.close();
+                throw new RuntimeException(new SQLException("ResultSet thread was interrupted"));
```

Comment:
include InterruptedException in the exc chain ?

Reply:
The outer `RuntimeException` here gets unwrapped so that we throw the `SQLException` directly. Do you want me to add `InterruptedException` to the cause of `SQLException`? I'm fine doing that if you think it's useful, though it seems a bit strange to generate a synthetic one here (would have the same stack trace as the `SQLException`).

Reply:
I'm going to leave this alone, since it's the same as the previous behavior. If we want to add it as a cause, we can do that separately later.